core/src/main/java/com/adobe/aem/commons/assetshare/content/properties/impl/SmartTagsImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Asset Share Commons
+ *
+ * Copyright (C) 2017 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.adobe.aem.commons.assetshare.content.properties.impl;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+import com.adobe.aem.commons.assetshare.content.properties.AbstractComputedProperty;
+import com.adobe.aem.commons.assetshare.content.properties.ComputedProperty;
+import com.day.cq.dam.api.Asset;
+
+/**
+ * This class would be responsible to generate the computed property list for the smart tags.
+ *  
+ * @author Tuhin Ghosh on 03/07/2018
+ *
+ */
+
+@Component(service = ComputedProperty.class)
+@Designate(ocd = SmartTagsImpl.Cfg.class)
+public class SmartTagsImpl extends AbstractComputedProperty<List<String>> {
+    public static final String LABEL = "Smart Tags";
+    public static final String NAME = "smartTags";
```

Comment:
Maybe `smartTagTitles` to align w existing `tagTitles` ?

Reply:
Sure thing